OrbitGl/TracepointTrack.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TracepointTrack.h"
+
+#include "App.h"
+
+TracepointTrack::TracepointTrack(TimeGraph* time_graph) : Track(time_graph) {
+  color_ = Color(0, 255, 0, 255);
+}
+
+void TracepointTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick, PickingMode) {
+  Batcher* batcher = &time_graph_->GetBatcher();
+  const TimeGraphLayout& layout = time_graph_->GetLayout();
+  float z = GlCanvas::kZValueEvent;
+  float track_height = layout.GetEventTrackHeight();
+
+  ScopeLock lock(GOrbitApp->GetCaptureData().GetTracepointEventBufferMutex());
+
+  std::map<uint64_t, orbit_client_protos::TracepointEventInfo>& tracepoints =
+      GOrbitApp->GetCaptureData().GetTracepointEventBufferTracepoints().at(thread_id_);
+
+  const Color kWhite(255, 255, 255, 255);
+
+  for (auto& pair : tracepoints) {
```

Comment:
The tracepoints are sorted by time so there is no need to loop over all the events. Search for the first > min_tick and the last < max_tick and loop between those.
There is something like map::lower_bound/upper_bound that should do.

Reply:
Yes, I will that to loop from 'tracepoints.lower_bound(min_tick)' and stop when I find an event that has the time larger than max_tick. Thank you for the suggestion!